rest/api_test.go

```diff
@@ -3137,6 +3137,55 @@ func TestDocSyncFunctionExpiry(t *testing.T) {
 	log.Printf("value: %v", value)
 }
 
+func TestCustomCookieName(t *testing.T) {
+
+	var rt RestTester
+	defer rt.Close()
+
+	// In order to test auth, you must disable admin party, which is set by default (should this really be set by default!?)
+	rt.noAdminParty = true
+
+	customCookieName := "TestCustomCookieName"
+	rt.DatabaseConfig = &DbConfig{
+		Name:              "db",
+		SessionCookieName: customCookieName,
+	}
+
+	// Disable guest user
+	dbcontext := rt.GetDatabase()
+	guestUser, err := dbcontext.Authenticator().GetUser("")
+	assertNoError(t, err, "Unexpected error")
+	guestUser.SetDisabled(true)
+
+	response := rt.SendAdminRequest("POST", "/db/_user/", `{"name":"user1", "password":"1234"}`)
+	assertStatus(t, response, 201)
+
+	resp := rt.SendAdminRequest("POST", "/db/_session", `{"name": "user1"}`)
+	assert.Equals(t, resp.Code, 200)
+
+	// Validate session response
+	sessionResponse := struct {
+		SessionID  string `json:"session_id"`
+		Expires    string `json:"expires"`
+		CookieName string `json:"cookie_name"`
+	}{}
+	err = json.Unmarshal(resp.Body.Bytes(), &sessionResponse)
+	assertNoError(t, err, "Unexpected error")
+	assert.Equals(t, sessionResponse.CookieName, customCookieName)
```

Comment:
We're still not validating that the Set-Cookie on a user session request is correctly modified.  What do you think about adding something like we're doing here: https://github.com/couchbase/sync_gateway/blob/master/rest/api_test.go#L1445

Reply:
Fixed.  Had to switch from `rt.SendAdminRequest()` -> `rt.SendRequest()`, which changed the response format quite a bit.